src/NodeVisitor/DynamicLoader.php

```diff
@@ -17,16 +17,146 @@ class DynamicLoader extends NodeVisitorAbstract
 {
     public $definitionCollector;
     public $prettyPrinter;
+    public $definitionResolver;
+    
+    private $collectAutoload;
```

Comment:
I suggest compute isAutoload inside constructor than let caller pass to it, might make code cleaner.